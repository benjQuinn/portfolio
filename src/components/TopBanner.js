import React, { useState, useEffect } from "react";
import moment from "moment";
import "../styles/top-banner.css";

function TopBanner() {
  const date = moment().format("ddd D MMM");
  const [time, setTime] = useState(moment().format("kk:mm/ss"));

  useEffect(() => {
    setInterval(() => {
      setTime(moment().format("kk:mm/ss"));
    }, 1000);
  }, []);

  return (
    <div className="top-banner">
      <span>P100</span>
      <span>100 CEEFAX</span>
      <span>{date}</span>
      <span>{time}</span>
    </div>
  );
}

export default TopBanner;
