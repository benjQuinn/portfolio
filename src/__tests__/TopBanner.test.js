import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import moment from "moment";
import TopBanner from "../components/TopBanner";

describe("TopBanner", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<TopBanner />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the ceefax text", () => {
    render(<TopBanner />);

    expect(screen.getByText(/p100/i)).toBeInTheDocument();
    expect(screen.getByText(/ceefax/i)).toBeInTheDocument();
  });

  it("renders the date correctly", () => {
    const date = moment().format("ddd D MMM").toUpperCase();

    render(<TopBanner />);

    expect(screen.getByText(date)).toBeInTheDocument();
  });

  it("renders the time correctly", () => {
    const time = moment().format("kk:mm/ss");

    render(<TopBanner />);

    expect(screen.getByText(time)).toBeInTheDocument();
  });
});
