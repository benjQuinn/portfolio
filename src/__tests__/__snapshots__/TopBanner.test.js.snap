// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBanner renders correctly 1`] = `
<DocumentFragment>
  <div
    class="top-banner"
  >
    <span>
      P100
    </span>
    <span>
      100 CEEFAX
    </span>
    <span>
      Sun 11 Sep
    </span>
    <span>
      18:17/01
    </span>
  </div>
</DocumentFragment>
`;
